src/app/component/Header/Header.style.scss

```diff
@@ -377,4 +377,25 @@
             }
         }
     }
+    &-CurrencySwitcher{
+        .currency{
+            min-width: 80px;
+            margin-right: 5px;
+            .Field{
+                margin-top: 0;
+            }
+        }
+        select{
+            padding: 10px;
+            margin-right: 5px;
+        }
+        @include mobile {
+            width: 0;
+            opacity: 0;
+            &_isVisible {
+                opacity: 1;
+                width: auto;
+            }
+        }
+    }
```

Comment:
In ideal condition there must be nesting of max one class (only building class-name). Do not nest classes in classes. This is anti-pattern and is not escapable by using BEM properly.